src/lib/pythongen.ml

```diff
@@ -213,10 +213,12 @@ let rec typecheck : type a. a typ -> string -> t list =
         variants
     in
     let check_contents =
-      List.fold_left
-        (fun acc x -> List.concat [ acc; check false x ])
-        (check true (List.hd variants_to_check))
-        (List.tl variants_to_check)
+      match variants_to_check with
+      | [] -> []
```

Comment:
@Vincent-lau I think @mseri meant that you didn't need the `| [] -> []` if you used `| v :: vs ->` form

Reply:
You need both, the second one pattern matches only on nonempty lists